torch/cuda/comm.py

```diff
@@ -77,9 +81,11 @@ def reduce_add(inputs, destination=None):
     if destination is None:
         destination = torch.cuda.current_device()
     input_size = inputs[0].size()
+    is_sparse = inputs[0].is_sparse
     nccl_root = None
     for i, inp in enumerate(inputs):
         assert inp.is_cuda, "reduce_add expects all inputs to be on GPUs"
+        assert inp.is_sparse == is_sparse, "reduce_add expects inputs to be either all sparse or all dense"
```

Comment:
I'm not really sure why do we need to enforce this. The function should be able to handle an arbitrary list of tensors

Reply:
I'll remove this.